src/settings/window_geometry.rs

```diff
@@ -1,16 +1,63 @@
-#[derive(Debug, Clone)]
+use crate::renderer::Renderer;
+use crate::settings::SETTINGS;
+use crate::window::WindowSettings;
+use glutin::dpi::PhysicalSize;
+use serde::{Deserialize, Serialize};
+use std::path::PathBuf;
+
+#[derive(Debug, Clone, Copy, Serialize, Deserialize)]
 pub struct WindowGeometry {
     pub width: u64,
     pub height: u64,
 }
 
+#[cfg(unix)]
+const SETTINGS_PATH_EXTENSION: &str = ".local/share/nvim/neovide-settings.json";
+#[cfg(windows)]
+const SETTINGS_PATH_EXTENSION: &str = "AppData/Local/nvim-data/neovide-settings.json";
+
+fn neovim_std_datapath() -> PathBuf {
+    let mut settings_path = dirs::home_dir().unwrap();
+    settings_path.push(SETTINGS_PATH_EXTENSION);
+    settings_path
+}
+
+pub fn try_to_load_last_window_size() -> Result<WindowGeometry, String> {
+    let settings_path = neovim_std_datapath();
+    let serialized_size = std::fs::read_to_string(&settings_path).map_err(|e| e.to_string())?;
+
+    let de: WindowGeometry = serde_json::from_str(&serialized_size).map_err(|e| e.to_string())?;
```

Comment:
nit: although it makes sense that de represents deserialized value. I try to avoid single or double character variable names. Its not that much more effort to write it out, and I think it makes the codebase more approachable.

Reply:
kk will fix